calicoctl/calico_ctl/pool.py

```diff
@@ -122,19 +122,61 @@ def ip_pool_add(cidrs, version, ipip, masquerade):
         print "IP in IP not supported for IPv6 pools"
         sys.exit(1)
 
-    # TODO Reject any cidrs that overlap with existing cidrs in the pool
-    pools = []
+    current_pool_nets = [pool.cidr for pool in client.get_ip_pools(version)]
+    new_pools = []
+
+    # Validate pool CIDRs
     for cidr in cidrs:
+        cidr_net = IPNetwork(cidr)
```

Comment:
It's not invalid to pass in `10.10.10.10/24`, so I don't think we should fail it.

Also: `IPNetwork("10.10.10.10/24") == IPNetwork("10.10.10.0/24")`